### PR TITLE
Updates: OTPClient 3.1.8, libcotp 2.0.1

### PR DESCRIPTION
**Pull request by realcharmer**
The new library version is needed for *OTPClient*, since it replaces `libbaseencode` in new releases.
I am also willing to adopt *OTPClient*, but it has a listed maintainer.

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures:
  - i686

### CODE REVIEW

**Comment by realcharmer** [comment]
Rebased to current master, bumped `OTPclient` to version `3.1.7` and `libcotp` to `2.0.1`.
Still waiting for review and merge.

File: srcpkgs/libcotp/template [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+maintainer="Emil Miler <em@0x45.cz>"
 license="Apache-2.0"
 homepage="https://github.com/paolostivanin/libcotp"
 distfiles="https://github.com/paolostivanin/libcotp/archive/v${version}.tar.gz"
-checksum=78dab6a2ee08e73f1d052dcb7c1ad069cc37fdf600f3f660d8e6299e11218f0b
+checksum=b111d528bbde7c1a0a392f49293b25ae33e6e78fbcbe378e0cf8bc6d59743d11
 
 libcotp-devel_package() {
-	depends="libcotp-${version}_${revision}"
+	depends="${sourcepkg}>=${version}"
```
Comment (classabbyamp) [inline srcpkgs/libcotp/template:16]:
```suggestion
	depends="${sourcepkg}>=${version}_${revision}"
```

**Review by classabbyamp — COMMENTED** [review]
you can adopt OTPClient if you would like, Ulf has not contributed since 2021

**Comment by realcharmer** [comment]
Alright, I adopted `OTPClient` as well and updated it to latest upstream version `3.1.8`.

**Pull request merged**